Src/LinearSolvers/MLMG/AMReX_MLNodeLap_3D_K.H

```diff
@@ -1135,6 +1135,263 @@ void mlndlap_gauss_seidel_aa (Box const& bx, Array4<Real> const& sol,
+        	    } 
+		    else 
+		    {
+			Real s0 = (-4.0)*fxyz*(sig(i-1,j-1,k-1)+sig(i,j-1,k-1)+sig(i-1,j,k-1)+sig(i,j,k-1)
+                                             + sig(i-1,j-1,k  )+sig(i,j-1,k  )+sig(i-1,j,k  )+sig(i,j,k  ));
+
+		        Real Ax = fxyz*(sol(i-1,j-1,k-1)*sig(i-1,j-1,k-1)
+                   	              + sol(i+1,j-1,k-1)*sig(i  ,j-1,k-1)
+                      	              + sol(i-1,j+1,k-1)*sig(i-1,j  ,k-1)
+                                      + sol(i+1,j+1,k-1)*sig(i  ,j  ,k-1)
+                                      + sol(i-1,j-1,k+1)*sig(i-1,j-1,k  )
+                                      + sol(i+1,j-1,k+1)*sig(i  ,j-1,k  )
+                                      + sol(i-1,j+1,k+1)*sig(i-1,j  ,k  )
+                                      + sol(i+1,j+1,k+1)*sig(i  ,j  ,k  ))
+                	      + fmx2y2z*(sol(i  ,j-1,k-1)*(sig(i-1,j-1,k-1)+sig(i,j-1,k-1))
+                               	      + sol(i  ,j+1,k-1)*(sig(i-1,j  ,k-1)+sig(i,j  ,k-1))
+                                      + sol(i  ,j-1,k+1)*(sig(i-1,j-1,k  )+sig(i,j-1,k  ))
+                                      + sol(i  ,j+1,k+1)*(sig(i-1,j  ,k  )+sig(i,j  ,k  )))
+                	      + f2xmy2z*(sol(i-1,j  ,k-1)*(sig(i-1,j-1,k-1)+sig(i-1,j,k-1))
+                         	       + sol(i+1,j  ,k-1)*(sig(i  ,j-1,k-1)+sig(i  ,j,k-1))
+                                       + sol(i-1,j  ,k+1)*(sig(i-1,j-1,k  )+sig(i-1,j,k  ))
+                                       + sol(i+1,j  ,k+1)*(sig(i  ,j-1,k  )+sig(i  ,j,k  )))
+                	      + f2x2ymz*(sol(i-1,j-1,k  )*(sig(i-1,j-1,k-1)+sig(i-1,j-1,k))
+                         	       + sol(i+1,j-1,k  )*(sig(i  ,j-1,k-1)+sig(i  ,j-1,k))
+                                       + sol(i-1,j+1,k  )*(sig(i-1,j  ,k-1)+sig(i-1,j  ,k))
+                         	       + sol(i+1,j+1,k  )*(sig(i  ,j  ,k-1)+sig(i  ,j  ,k)))
+                	      + f4xm2ym2z*(sol(i-1,j,k)*(sig(i-1,j-1,k-1)+sig(i-1,j,k-1)+sig(i-1,j-1,k)+sig(i-1,j,k))
+                           	       + sol(i+1,j,k)*(sig(i  ,j-1,k-1)+sig(i  ,j,k-1)+sig(i  ,j-1,k)+sig(i  ,j,k)))
+		              + fm2x4ym2z*(sol(i,j-1,k)*(sig(i-1,j-1,k-1)+sig(i,j-1,k-1)+sig(i-1,j-1,k)+sig(i,j-1,k))
+                	               + sol(i,j+1,k)*(sig(i-1,j  ,k-1)+sig(i,j  ,k-1)+sig(i-1,j  ,k)+sig(i,j  ,k)));
+
+			a_ls(k-lo.z) = fm2xm2y4z*(sig(i-1,j-1,k-1)+sig(i,j-1,k-1)+sig(i-1,j,k-1)+sig(i,j,k-1));
+            		b_ls(k-lo.z) = s0;
+            		c_ls(k-lo.z) = fm2xm2y4z*(sig(i-1,j-1,k  )+sig(i,j-1,k  )+sig(i-1,j,k  )+sig(i,j,k  ));
+            		u_ls(k-lo.z) = 0.;  
+            		r_ls(k-lo.z) = rhs(i,j,k) - Ax;
+				
+		    }
+		}
+		tridiagonal_solve(a_ls, b_ls, c_ls, r_ls, u_ls, gam, ilen);
+
+		for (int k = lo.z; k <= hi.z; ++k)
+		{   
+	      	    sol(i,j,k) = u_ls(k-lo.z);
+		}
+	    }
+        }
+    }
+    else if (idir == 1)
+    {
+        for (int i = lo.x; i <= hi.x; ++i)
+        {
+            AMREX_PRAGMA_SIMD
```

Comment:
This should be removed.  Also are there any reasons the i-loop is outside the k-loop?  Generally speaking we want k-loop to be outside i-loop because the arrays are in Fortran order.

Reply:
If there are no reasons for the loop order chosen, please switch the orders in other places too.